File: DocumentAnalysisSystem/Models/Section.cs
using System.Collections.Generic;
using DocumentAnalysisSystem.Visitors;

namespace DocumentAnalysisSystem.Models
{
    public class Section : DocumentComponent
    {
        private readonly List<DocumentComponent> _children = new();

        public Section(string title) : base(title)
        {
            Title = title;
        }

        public string Title { get; }

        public override IReadOnlyList<DocumentComponent> Children => _children.AsReadOnly();

        public void Add(DocumentComponent component)
        {
            _children.Add(component);
        }

        public override void Accept(IDocumentVisitor visitor)
        {
            visitor.Visit(this);
            foreach (var child in _children)
            {
                child.Accept(visitor);
            }
        }
    }
}

File: DocumentAnalysisSystemJava/src/main/java/com/documentanalysis/models/Section.java
package com.documentanalysis.models;

import com.documentanalysis.visitors.IDocumentVisitor;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Section extends DocumentComponent {
    private final String title;
    private final List<DocumentComponent> children = new ArrayList<>();

    public Section(String title) {
        super(title);
        this.title = title;
    }

    public String getTitle() {
        return title;
    }

    @Override
    public List<DocumentComponent> getChildren() {
        return Collections.unmodifiableList(children);
    }

    public void add(DocumentComponent component) {
        children.add(component);
    }

    @Override
    public void accept(IDocumentVisitor visitor) {
        visitor.visit(this);
        for (DocumentComponent child : children) {
            child.accept(visitor);
        }
    }
}
